front/src/modules/timetableItem/helpers/formatTimetableItemWithDetails.ts

```diff
@@ -16,26 +18,21 @@ import { Duration } from 'utils/duration';
 import { jouleToKwh } from 'utils/physics';
 import { formatKmValue } from 'utils/strings';
 
-const formatSummary = (
-  timetableItem: TimetableItem,
-  summary?: SimulationSummaryResult
-): SimulationSummary | undefined => {
-  // not loaded
-  if (!summary) {
-    return undefined;
-  }
+import { extractOccurrenceDetailsFromPacedTrain } from './pacedTrain';
 
-  // invalid train
-  if (summary.status !== 'success')
-    return {
-      isValid: false,
-      invalidReason:
-        summary.status === 'pathfinding_not_found' || summary.status === 'pathfinding_input_error'
-          ? summary.error_type
-          : summary.status,
-    };
+const extractInvalidReason = (summary: SimulationSummaryResult) => {
+  if (summary.status === 'success') {
+    throw new Error();
+  }
+  return summary.status === 'pathfinding_not_found' || summary.status === 'pathfinding_input_error'
+    ? summary.error_type
+    : summary.status;
+};
```

Comment:
Both times where we call this function, we check for summary.status === 'success', is it really necessary to add another check here ?

We could rather do : 
```suggestion
const extractInvalidReason = (summary: Exclude<SimulationSummaryResult, { status: 'success' }>) =>
  summary.status === 'pathfinding_not_found' || summary.status === 'pathfinding_input_error'
    ? summary.error_type
    : summary.status;
```

Reply:
You're right, thanks :)